chore: Rebase `avahi_name_conflict` commits onto `edge`

SyntaxColoring opened the pull request:
# Overview

This PR is to rebase two commits atop `edge`:

1. One commit for the entirety of #10523.
2. One commit for the entirety of #10559.
3. One fixup commit to change `sleep(3600)` to `sleep(10)`. See discussion below in this PR.

Fixes bug #10126.

# Review requests

* We agree with the plan to click the GitHub "rebase and merge" button.
* We agree with merging this into `edge` (as opposed to an alpha or beta branch or something).
* No unexpected commits are in here.

# Risk analysis

This should be safe at this point because:

* This has already been discussed, code-reviewed, and tested in #10559.
* The only other change to `update-server` in the meantime has been code linting changes, #10695.

SyntaxColoring committed:
refactor(update-server): Split `name_management` into submodules (#10523)

SyntaxColoring committed:
fix(update-server): Keep name deconflicted with other devices on the network (#10559)

Fixes #10126.

mcous reviewed (approved):
Let's get a 👍 from the embedded team, too

sfoster1 commented on update-server/otupdate/common/name_management/avahi.py:
```diff
@@ -137,3 +50,280 @@ async def restart_daemon() -> None:
+    (If we care to fix this, we should look into preserving grapheme clusters:
+    https://unicode.org/reports/tr29/)
+    """
+    encoded = s.encode("utf-8")[:maximum_utf8_octets]
```
I think this won't be as much of an issue because our name acceptability regex may prevent characters like this, depending on how the engine is implemented. it really is pretty gross though, and I think there are better ways to do it - see top level review

SyntaxColoring replied:
The only name acceptability regex that I know of today is client-side. Is that the one that you mean?

If so, yes, it'll prevent non-ASCII. But we intend to loosen that eventually. I figured we may as well try to handle things nicely now in case a client that's looser with names ever talks to this server version.

Agreed with your top-level review about this being gross and there being better ways.

sfoster1 commented on update-server/otupdate/common/name_management/avahi.py:
```diff
@@ -137,3 +50,280 @@ async def restart_daemon() -> None:
+
+        # "3" comes from AVAHI_ENTRY_GROUP_COLLISION being index 3 in this enum:
+        # https://github.com/lathiat/avahi/blob/v0.8/avahi-common/defs.h#L234
+        avahi_entry_group_collision = dbus.Int32(3)
```
[nit] love the comment with the source, but maybe also make this a class constant so you can see it from the code alone?

SyntaxColoring replied:
Yeah, @X-sam suggested this in https://github.com/Opentrons/opentrons/pull/10559#discussion_r888179587 too. Two people independently suggesting this probably means it's a good idea, so I will.

SyntaxColoring replied:
Oops, it turns out that this is hard because dev machines don't have the `dbus` module.

This syntax:

```python
class _SyncClient:
    _AVAHI_ENTRY_GROUP_COLLISION = dbus.Int32(3)
    ...
```

Will evaluate `dbus.Int32(3)` at class definition time, and run into a `NameError` because `dbus` isn't defined.

I don't see an easy way of resolving this without doing something like guarding the whole class definition, which, anecdotally, I've seen mypy have trouble with.

Leaving this alone for now.

sfoster1 commented on update-server/otupdate/common/run_application.py:
```diff
@@ -0,0 +1,37 @@
+            # It seems like there should be a better way of doing this,
+            # but this is what the docs recommend.
+            # https://docs.aiohttp.org/en/stable/web_advanced.html#application-runners
+            await asyncio.sleep(3600)
```
can we make this like `await asyncio.sleep(10)`? i get why you would want to do this, but either
- it doesn't matter how long you sleep as long as it's "a while" (and ten seconds counts)
- something weird happens when this task wakes and it'll be easier to notice if it's every ten seconds instead of every hour

even better honestly would be establishing some dummy quitflag, setting it, and then await on it dropping. will never wake accidentally and provides a pretty clear path to adding graceful quitting.

SyntaxColoring replied:
Good thoughts.

Going with `await asyncio.sleep(10)` for now because it's a more minimal change and it keeps us mirroring [the aiohttp docs](https://docs.aiohttp.org/en/stable/web_advanced.html#application-runners). But the dummy quitflag is a clever solution, I think I agree that it's better, and I'll make a note to do that in another PR.

sfoster1 replied:
if you're gonna make a note for that - maybe note to go ahead and implement actual quitting by handling `sigterm` (default for systemd process killing, could change to basically anything else if you wanted to [in the unit file](https://www.freedesktop.org/software/systemd/man/systemd.kill.html#)) and raising the flag,

sfoster1 reviewed (commented):
The truncation stuff I think can have some slightly surprising outcomes. A better way to do this might be to limit the length of the names you can set low enough to give enough space for, say, 4 or 5 or 6 digits of service name deconfliction, and wrap the sequence numbers at that limit. Then there's no dynamic truncation stuff required. It also solves the problem with losing the original basename so that you might end up with just "Num0" as your name after a while.

SyntaxColoring commented:
> The truncation stuff I think can have some slightly surprising outcomes. A better way to do this might be to limit the length of the names you can set low enough to give enough space for, say, 4 or 5 or 6 digits of service name deconfliction, and wrap the sequence numbers at that limit. Then there's no dynamic truncation stuff required. It also solves the problem with losing the original basename so that you might end up with just "Num0" as your name after a while.

Yeah, that definitely would have been better.

I think I'd like to stick with the logic as written for now, because:

* It's already been validated.
* The only reason we need to implement this ourselves in the first place is because of the app-side bug #10672. I'm hoping we can fix that, [let Avahi pick the alternative name for us](https://www.avahi.org/doxygen/html/alternative_8h.html#ae0b973d05cf10ee64ff755cff0b85d14), and rip out this code.

sfoster1 reviewed (approved):
Based on discussion here, sounds good to me!

Will say though, I'm not an expert, but if splitting grapheme clusters is a concern I don't know [how much I want to leave that to avahi](https://github.com/lathiat/avahi/blob/d1e71b320d96d0f213ecb0885c8313039a09f693/avahi-common/utf8.c)

SyntaxColoring committed:
refactor(update-server): Busy-sleep for 10s instead of 1h (#10706)

Addresses review comment:
https://github.com/Opentrons/opentrons/pull/10706#discussion_r893915850
